### ISSUE TITLE
Dicas para iniciantes no FRONT-END:O que estudar e não estudar?

### ISSUE THREAD

**Issue by lucasmoore2k** [frontendbr/forum#1680]
Olá galera é o seguinte:esse ano resolvi me dedicar em aprender de vez programação para front end, contudo me deparei com várias discussões sobre o que é essencial,exemplo vi gente dizendo que jQuery e Ajax você não precisa aprender(pelo menos não de forma aprofundada),PHP não precisa usar as frameworks(?).

Eu sei que de primeira é meio impossível aprender tantas coisas e tantos frameworks num curto período de tempo(2 meses) ,por isso quero ter um estudo o mais otimizado possível,a profundidade e os detalhes eu pegaria durante o trabalho e vivência na linguagem.

Já que no momento não trabalho e gostaria de aprender  pelo menos o essencial para começar a fazer projetos free lance ou tentar pleitear uma vaga como junior.

*Html,CSS e Javascript eu já vi,só falta mais prática e aprender os frameworks.

Agradeço!

**Comment by ojvribeiro** [comment]
Recomendo estudar frameworks JavaScript como o [React](https://pt-br.reactjs.org/) ou [Vue](https://vuejs.org/), pré-processadores de CSS como o [Sass](https://sass-lang.com/) ou o [Less](http://lesscss.org/) e aprender um pouco sobre o ES6, que é o "JavaScript de hoje".

Pra aulas, motivação e conselhos sobre o mundo do desenvolvimento web, recomendo os canais do [Rocketseat](https://www.youtube.com/channel/UCSfwM5u0Kce6Cce8_S72olg) e do [Filipe Deschamps](https://www.youtube.com/channel/UCU5JicSrEM5A63jkJ2QvGYw) no YouTube.

**Comment by portothree** [comment]
Não partilho da mesma opnião do @ojvribeiro, não penso que já seja uma boa ir para frameworks.

HTML: Já consegue com facilidade organizar a sua estrutura de forma semântica? 
CSS: Já sabe como tirar proveito do Cascading? Já consegue definir breakpoints e partir de uma abordagem Mobile-first?
JavaScript puro: Já entende como está a manipular a API de DOM e como acontece o event bubbling? 

Se você **sentir** a necessidade de usar um framework depois de alguns projetos com javascript puro, vai ser entendendo qual a solução o framework está a trazer para o seu projeto e sem pular passos importantes. :)

**Comment by lucasmoore2k** [comment]
> Não partilho da mesma opnião do @ojvribeiro, não penso que já seja uma boa ir para frameworks.
> 
> HTML: Já consegue com facilidade organizar a sua estrutura de forma semântica?
> CSS: Já sabe como tirar proveito do Cascading? Já consegue definir breakpoints e partir de uma abordagem Mobile-first?
> JavaScript puro: Já entende como está a manipular a API de DOM e como acontece o event bubbling?
> 
> Se você **sentir** a necessidade de usar um framework depois de alguns projetos com javascript puro, vai ser entendendo qual a solução o framework está a trazer para o seu projeto e sem pular passos importantes. :)

Acho que o que os dois falaram está correto,mas a longo prazo essa escolha de aprender primeiro as bases do Front End por assim dizer vai facilitar muito minha vida quando estudar frameworks.
Inclusive já vi vagas em que eles pedem os frameworks,mas eles queriam alguém que não dependesse deles.
Eu estou seguindo uma linha de estudos de focar nos fundamentos e em paralelo os frameworks,mas de forma não muito aprofundada somente para ter uma noção com isso quando eu partir realmente para o aprendizado deles eu já não estaria sem terreno.

**Comment by carlosviniciusananias** [comment]
Creio que inicialmente deve-se dominar **html**, **css** e **javascript** puro.

Dominando o 'arroz com feijão' você parte para os famosos frameworks. 

Uma coisa que faço sempre antes de iniciar os estudos em determinada tecnologia é me perguntar: _O quanto ela é relevante? Há necessidade de profissionais capacitados no mercado? Qual curva de aprendizado?_

**Comment by ninetails** [comment]
Pessoalmente acho que falta um objetivo. "Aprender frontend" ainda é muito amplo e são grandes as chances de se perder no caminho ainda mais com um curto espaço de tempo.

Sorteia um cara daqui: https://github.com/felipefialho/frontend-challenges e case com o que vc quer aprender. E acho que sobre o que aprender e o que não aprender, primeiro faça funcionar. Depois vc refatora e/ou mostra o código pra alguém que possa te _mentorar_ para ver o que dá pra melhorar. E daí vai adicionando _handicaps_: tirar tabelas, tirar jQuery se usar, usar menos divs, melhorar usabilidade e acessibilidade........

**Comment by danilosilvadev** [comment]
Vue é o mais fácil de começar, vale a pena talvez. React tem mais vaga(por enquanto). E Stephen Gridder é o instrutor do Udemy q ensina td isso de maneira maravilhosa.

**Comment by wilsonneto-dev** [comment]
concordo com o @danilosilvadev... exceto na parte do "por enquanto" kk

**Comment by ojvribeiro** [comment]
Não entendi o porquê dos downvotes. O @lucasmoore2k mesmo disse:

> *Html,CSS e Javascript eu já vi,só falta mais prática e aprender os frameworks.

**Comment by karlFrascari2** [comment]
Pessoal, existe um `Roamap` muito bom que ficou muito completo, que utilizo até hoje. 
Ele vai desdo básico do básico `Html, css e Javascript` até `Aplicações mobile com Javascript` :

[Amazing Front-end Roadmap](https://roadmap.sh/frontend)

![image](https://user-images.githubusercontent.com/9298260/74591448-e160fb80-4ff6-11ea-97ec-79059405fb4c.png)


Em cima disso estou trabalhando em destrinchar esse road map em um conteudo que seja mais facil encontrar informações sobre esses tópicos: 


##  **[LINK: Front-end Learning Path 2020](https://github.com/karlFrascari2/front-end-learning-path)**
<br/> 

> ![image](https://user-images.githubusercontent.com/9298260/74591737-30a82b80-4ff9-11ea-8d6d-6a35ded45ec8.png)



Pode ser que te ajude @lucasmoore2k ! Valeu!
